src/components/ContextBar.test.js

```diff
@@ -0,0 +1,625 @@
+
+  it('will render button with error status', () => {
+    const jupyterContextBarProps = {
+      ...contextBarProps,
+      runtimes: [
+        {
+          ...jupyter,
+          status: 'Error'
+        }
+      ],
+      persistentDisks: []
+    }
+
+    // Act
+    const { getByText, getByLabelText } = render(h(ContextBar, jupyterContextBarProps))
+
+    //Assert
+    expect(getByText('Rate:'))
+    expect(getByText(`${Utils.formatUSD(RUNTIME_COST)}`))
+    expect(getByLabelText('Environment Configuration'))
+    expect(getByLabelText(new RegExp(/Jupyter Environment/i)))
+    expect(getByText(/Error \$0.00\/hr/))
+  })
+})
+
+describe('ContextBar - actions', () => {
+  it('clicking environment configuration opens CloudEnvironmentModal', () => {
+    // Act
+    const { getByText, getByLabelText } = render(h(ContextBar, contextBarProps))
+
+    const envConf = getByLabelText('Environment Configuration')
+    fireEvent.click(envConf)
+
+    // Assert
+    expect(getByText('Cloud Environment Details'))
+  })
+  it('clicking Jupyter opens CloudEnvironmentModal with Jupyter as filter for tool.', () => {
+    // Arrange
+    const jupyterContextBarProps = {
+      ...contextBarProps,
+      runtimes: [jupyter],
+      persistentDisks: [runtimeDisk]
+    }
+
+    // Act
+    const { getByText, getByLabelText } = render(h(ContextBar, jupyterContextBarProps))
+    fireEvent.click(getByLabelText(new RegExp(/Jupyter Environment/i)))
+
+    // Assert
+    getByText('Cloud Environment Details')
+    getByText(tools.Jupyter.label)
+  })
+  it('clicking Galaxy opens CloudEnvironmentModal with Galaxy as filter for tool.', () => {
+    // Arrange
+    const galaxyContextBarProps = {
+      ...contextBarProps,
+      apps: [galaxyRunning],
+      appDataDisks: [galaxyDisk]
+    }
+
+    // Act
+    const { getByLabelText, getByText } = render(h(ContextBar, galaxyContextBarProps))
+    fireEvent.click(getByLabelText(new RegExp(/Galaxy Environment/i)))
+
+    // Assert
+    getByText('Cloud Environment Details')
+    getByText(tools.Galaxy.label)
+  })
+  it('clicking RStudio opens CloudEnvironmentModal with RStudio as filter for tool.', () => {
+    // Act
+    const rstudioContextBarProps = {
+      ...contextBarProps,
+      runtimes: [rstudioRuntime],
+      apps: [galaxyRunning],
+      appDataDisks: [galaxyDisk],
+      persistentDisks: [runtimeDisk]
+    }
+
+    // Act
+    const { getByText, getByLabelText } = render(h(ContextBar, rstudioContextBarProps))
+    fireEvent.click(getByLabelText(new RegExp(/RStudio Environment/i)))
+
+    // Assert
+    getByText('Cloud Environment Details')
+    getByText(tools.RStudio.label)
+  })
+
+  it('clicking Terminal will attempt to start currently stopped runtime', () => {
+    // Arrange
+    global.window = Object.create(window)
+    const url = 'http://dummy.com'
+    Object.defineProperty(window, 'location', {
+      value: {
+        href: url
+      },
+      writable: true,
+      hash: '/'
+    })
```

Comment:
It might be worth adding a function to Nav that wraps access to `window.location` to make mocking easier.